dig.go

```diff
@@ -753,7 +753,7 @@ func isFieldOptional(f reflect.StructField) (bool, error) {
 func shallowCheckDependencies(c containerStore, p param) error {
 	var missing errMissingManyTypes
 	var addMissingNodes []*dot.Param
-	walkParam(p, paramVisitorFunc(func(p param) bool {
+	walkParam(p, newParamVisitOnce(make(map[string]struct{}), func(p param) bool {
```

Comment:
I don't think you need this here because there's no recursion on new
dependencies here. (The only time we have recursion here is when we have non-
leaf nodes: paramList, paramGroup.)

This checks the direct descendants of a parameter so checking for repeated
visits of nodes will be beneficial only if you had `func(io.Reader,
io.Reader)` or similar.